tests/CMakeLists.txt

```diff
@@ -1,8 +1,8 @@
 
 add_library(test_common_libs INTERFACE)
-target_link_libraries(test_common_libs INTERFACE stdc++fs gtest gtest_main)
+target_link_libraries(test_common_libs INTERFACE pthread stdc++fs GTest::GTest GTest::Main)
```

Comment:
`CMake` provides a GTest wrapper, did you try it?

Reply:
From what I see, the way to build is still linking `gtest` and `gtest_main` the way I did it. CMake then provides `ctest` which lets you run them